src/components/SecondaryPanes/XHRBreakpoints.js

```diff
@@ -88,9 +101,21 @@ class XHRBreakpoints extends Component<Props, State> {
     e.preventDefault();
     e.stopPropagation();
 
-    this.props.setXHRBreakpoint(this.state.inputValue, "ANY");
+    const setXHRBreakpoint = function() {
+      this.props.setXHRBreakpoint(
+        this.state.inputValue,
+        this.state.inputMethod
+      );
+      this.hideInput();
+    };
 
-    this.hideInput();
+    // force update inputMethod in state for mochitest purposes
+    // before setting XHR breakpoint
+    this.setState(
+      // $FlowIgnore
+      { inputMethod: e.target.children[1].value },
+      setXHRBreakpoint
```

Comment:
Umm, we are setting this `setXHRBreakpoint` in the state, I don't understand this part.

Reply:
We are. This is just a workaround for mochitests to make it testable, since there's no reliable way to simulate a user change event for the `<select>` element without importing `ReactUtils`.

Instead, I just programmatically change the value of the `<select>` element. But changing the value itself doesn't trigger a change event, which doesn't update the `inputMethod` in state, so I added another "force update" on submit to make sure that it updates the `inputMethod` on submit.

`setState` just requests a state change to be queued - it doesn't always immediately update the state as it's called. `setXHRBreakpoint` is passed into `setState` specifically to make sure that it happens after the `inputMethod` is updated on submit. Otherwise, what happens is `setXHRBreakpoint` gets called first before the state is updated.

Reply:
👍 Makes sense now. Thanks :)